### ISSUE TITLE
[desgin consult problem] Is quickwit's performace is influenced by the coming indexing data sequence?

### ISSUE THREAD

**Issue by geek-frio** [quickwit-oss/quickwit#1859]
After have read the quickwit Indexer implementation code, I have a dout.
Assuming every coming record in my ingest dataset is not sort by datatime, but have a very large time range, like this::

2022/01/01 10:01:00 data1
2021/02/02 00:00:11 data2
1999/01/01 00:00:11 data3
2022/01/01 09:00:00 data4
...

In this case, every index split that quickwit generated will have a very large time range step,then when I do a dist query, every index split will be searched because the index split's time range is not useful any more. isn't it ?

**Comment by fmassot** [comment]
You're right in your analysis. If the timestamps are in random order then each split will have timestamps with a big time range. And a query will not be able to benefit from timestamp-based pruning.
It's obviously better if you don't mix too much the timestamps.

**Comment by geek-frio** [comment]
@fmassot thks!
